code/_onclick/click.dm

```diff
@@ -327,26 +327,35 @@
 		ML.pulled(src)
 
 /mob/living/CtrlClick(mob/user)
-	if(isliving(user) && Adjacent(user) && !user.incapacitated())
-		if(world.time < user.next_move)
-			return FALSE
-		var/mob/living/user_living = user
-		if(user_living.apply_martial_art(src, null, is_grab=TRUE) == MARTIAL_ATTACK_SUCCESS)
-			user_living.changeNext_move(CLICK_CD_MELEE)
-			return
-	else
-		..()
+	if(!isliving(user) || !Adjacent(user) || user.incapacitated())
+		return ..()
+
+	if(world.time < user.next_move)
+		return ..()
+
+	var/mob/living/user_living = user
+	if(user_living.apply_martial_art(src, null, is_grab=TRUE) == MARTIAL_ATTACK_SUCCESS)
+		user_living.changeNext_move(CLICK_CD_MELEE)
+		return TRUE
+
+	return ..()
 
 
 /mob/living/carbon/human/CtrlClick(mob/user)
-	if(ishuman(user) && Adjacent(user) && !user.incapacitated())
-		if(world.time < user.next_move)
-			return FALSE
-		var/mob/living/carbon/human/H = user
-		H.dna.species.grab(H, src, H.mind.martial_art)
+
+	if(!ishuman(user) ||!Adjacent(user) || user.incapacitated())
+		return ..()
+
+	if(world.time < user.next_move)
+		return ..()
+
+	var/mob/living/carbon/human/H = user
+	if(H.dna.species.grab(H, src, H.mind.martial_art))
 		H.changeNext_move(CLICK_CD_MELEE)
-	else
-		..()
+		return TRUE
+
+	return ..()
```

Comment:
Don't these new returns change behavior?

Reply:
You mean return TRUE in the if ?

Reply:
Well the old pattern was
If not human/adjacent/alive call parent
otherwise, if we're spamming clicks, return false.
if we successfully grab, return false

the current pattern is
if not human/adjacent/alive call parent
if we're spam clicking, call parent
if we grab successfully, return True
otherwise, return parent

Reply:
It makes more sense to return TRUE after the grab and let the subsequent procs that called it handle the result of it than return FALSE after an action that completed successfully.

Also I didn't find any changed behaviour but I'm not a QA team, so I might have missed something.

Reply:
spam clicking should return false, outside of that, I can live with this

Reply:
Done.